lib/park.rb

```diff
@@ -0,0 +1,35 @@
+class Park
+  attr_reader :name, :admission_price, :vehicles
+
+  def initialize(name, admission_price)
+    @name = name
+    @admission_price = admission_price
+    @vehicles = []
+  end
+
+  def add_vehicle(vehicle)
+    @vehicles << vehicle
+  end
+
+  def all_passengers
+    @vehicles.flat_map(&:passengers)
+  end
+
+  def revenue
+    all_passengers.count { |passenger| passenger.adult? } * @admission_price
+  end
+
+  # iteration 4 below 
+  
+  def passenger_names
+    all_passengers.map(&:name).sort # sorts names alphabetically
+  end
+
+  def minors
```

Comment:
similar to revenue, this functionality is good, but there might be a better way - instead of iterating, then iterating, then iterating (.select, .map, .sort), you could use each to filter and transform in one step and end up only iterating twice (.each, .sort):

```
minor_names = []
all_passengers.each do |passenger|
  if !passenger.adult?
    minor_names << passenger.name
  end
end
minor_names
```